test: add networkscenemanager additive scene loading tests

0xFA11 opened the pull request:
I managed to break PR #1030 — so, I'm fixing and putting this up for review again :)
It's owned by @NoelStephensUnity (sorry!)

---

This includes the unit testing for additive scene loading using multi-instances.
It includes a minor adjustment to NetworkSceneManager to handle unit testing with multi-instances.
This removes the legacy Runtime NetworkSceneManagerTest and migrates the check for assert when EnableSceneManagement is set to false into the new NetworkSceneManagerTests.

Current Tests:
- 1 host and 9 clients loading a scene additively and then unloading it in unit test mode.  This tests the scene event messaging and local notification pipelines for loading and unloading additive scenes.  (see [NetworkSceneManagerTests](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/blob/test/AdditiveSceneLoading/testproject/Assets/Tests/Runtime/NetworkSceneManagerTests.cs))
- Tests the various exception cases to assure if EnableSceneManagement is not set an exception is thrown or if a client attempts to start a load or unload scene event an exception is thrown
- Verifies that a scene event cannot be started during an already in progress scene event
- Verifies that trying to unload a scene not already loaded returns the appropriate failed unload status
- Verifies that trying to load a scene name for a scene that does not exist (is not registered) returns the appropriate failed to start scene event status.

0xFA11 committed:
redo #1030

MFatihMAR commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:
```diff
@@ -96,6 +96,10 @@ public class NetworkSceneManager
         private static bool s_IsSceneEventActive = false;
 
+#if UNITY_EDITOR || DEVELOPMENT_BUILD
+        private bool m_IsRunningUnitTest = SceneManager.GetActiveScene().name.StartsWith("InitTestScene");
```
I have a better idea for this :)

0xFA11 committed:
comment a todo for `m_IsRunningUnitTest`

0xFA11 commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:
```diff
@@ -96,6 +96,12 @@ public class NetworkSceneManager
         /// </summary>
         private static bool s_IsSceneEventActive = false;
 
+        // TODO: Remove `m_IsRunningUnitTest` entirely after we switch to multi-process testing
+        // In MultiInstance tests, we cannot allow clients to load additional scenes as they're sharing the same scene space / Unity instance.
```
in the previous PR #1030, @andrews-unity and @LukeStampfli approved this PR and I also do approve this PR too.
please feel free to approve your own PR @NoelStephensUnity and merge it in :)

NoelStephensUnity committed:
test project

Adding check for unit testing at the end of OnClientUnloadScene to determine if we need to just passthrough to the callback (for multiInstance unit testing only)

Fixing minor issue with NetworkObject not being around after unloading a scene for NetworkPrefabPool.

NoelStephensUnity reviewed (approved):
After the last minor adjustments, this PR is good to go.
